Source/RageCommunity.Library/Graphics/Checkpoint.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Drawing;
+using Rage;
+using Rage.Native;
+using RageCommunity.Library.Extensions;
+
+namespace RageCommunity.Library.Graphics
+{
+    /// <summary>
+    /// This class represents a checkpoint within GTA V.
+    /// </summary>
+    /// <remarks>
+    /// Originally created by PNWParksFan. 
+    /// </remarks>
+    public class Checkpoint : IDeletable, ISpatial
+    {
+        private readonly Boolean setToGround;
+        private Color color;
+        private Single height;
```

Comment:
When I open this in Visual Studio, it suggests simplifying certain value types such as `Single` and `Int32` to `float` and `int` respectively.  Should we do this for all value types where possible?

Reply:
This is a tricky one. Microsoft suggest to use CLR type because it is not language specific. Otherwise visual studio suggest the non-CLR type and I saw many RPH plugin devs like to use non-CLR type. I'm open to change these, but I'm not sure what the right decision is.

Reply:
I wasn't aware of CLR vs non-CLR typing, but this contribution is the first time I've seen CLR types used in a plugin.  I don't think there is a right or wrong decision, similar to the private field naming convention.  I guess the only thing is VS will nag you about simplifying the type.